java-frontend/src/main/java/org/sonar/java/se/checks/UnclosedResourcesCheck.java

```diff
@@ -92,19 +94,23 @@ public ProgramState checkPostStatement(CheckerContext context, Tree syntaxNode)
 
   @Override
   public void checkEndOfExecutionPath(CheckerContext context, ConstraintManager constraintManager) {
-    final List<ObjectConstraint> constraints = context.getState().getFieldConstraints(Status.OPENED);
-    for (ObjectConstraint constraint : constraints) {
-      Tree syntaxNode = constraint.syntaxNode();
-      String name = null;
-      if (syntaxNode.is(Tree.Kind.NEW_CLASS)) {
-        name = ((NewClassTree) syntaxNode).identifier().symbolType().name();
-      } else if (syntaxNode.is(Tree.Kind.METHOD_INVOCATION)) {
-        name = ((MethodInvocationTree) syntaxNode).symbolType().name();
-      }
-      if (name != null) {
-        context.reportIssue(syntaxNode, this, "Close this \"" + name + "\".");
-      }
+    ExplodedGraph.Node node = context.getNode();
+    context.getState().getValuesWithConstraints(Status.OPENED)
+      .forEach((sv, c) -> processUnclosedSymbolicValue(node, sv));
+  }
+
+  private void processUnclosedSymbolicValue(ExplodedGraph.Node node, SymbolicValue sv) {
+    List<JavaFileScannerContext.Location> flow = SECheck.flow(node, sv);
+    flow.stream()
+      .filter(loc -> loc.syntaxNode.is(Tree.Kind.NEW_CLASS) || loc.syntaxNode.is(Tree.Kind.METHOD_INVOCATION))
+      .forEach(loc -> reportIssue(loc.syntaxNode, String.format("Close this \"%s\".", name(loc.syntaxNode)), ImmutableSet.of(flow)));
```

Comment:
Something is odd here : for each element of the flow which is new class or method invocation you report an issue with the whole flow as flow for the issue ? 
This tends to demonstrate we probably need something to test the flow thoroughly  in the file.